presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -228,6 +231,32 @@ public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return deserialize(input).isEmpty();
     }
 
+    @Description("Returns true if the input geometry is well formed")
+    @ScalarFunction("ST_IsValid")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean stIsValid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        return GeometryJtsUtils.deserialize(input).isValid();
+    }
+
+    @Description("If input geometry is not valid, return a reason why. Otherwise, returns null.")
+    @ScalarFunction("geometry_is_valid_reason")
```

Comment:
This function returns a varchar/reason string, but the name `geometry_is_valid_reason` sounds like it returns a boolean (`is...`).